backend/main/src/Group/Query/FindContainersOfAUserQuery/FindContainersOfAUserQuery.ts

```diff
@@ -0,0 +1,94 @@
+import { IQuery } from "src/Shared/Layers/Query/IQuery";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import { ContainerData } from "src/Shared/Adapters/DB/NishikiDBTypes";
+import { Ok, Err, Result } from "result-ts-type";
+import { QueryError } from "src/Shared/Utils/Errors";
+import { isValidUUIDV4 } from "src/Shared/Utils/Validator";
+
+export class FindContainersOfAUserQuery
+	implements IQuery<{ userId: string }, IContainerData, InvalidUUId>
+{
+	constructor(private readonly nishikiDynamoDBClient: NishikiDynamoDBClient) {}
+
+	public async execute(input: { userId: string }): Promise<
+		Result<IContainerData, InvalidUUId>
+	> {
+		const { userId } = input;
+		if (isValidUUIDV4(userId)) {
+			return Err(new InvalidUUId("User ID is invalid"));
+		}
+
+		const userGroups =
+			await this.nishikiDynamoDBClient.listOfUsersGroup(userId);
+
+		const containersOfUser: IContainerData = { containers: [] };
+
+		for (const userGroup of userGroups) {
+			const [containerIds, groupData] = await Promise.all([
+				this.nishikiDynamoDBClient.listOfContainers(userGroup.groupId),
+				this.nishikiDynamoDBClient.getGroup({ groupId: userGroup.groupId }),
+			]);
```

Comment:
You shouldn't call DB inside the for loop, it could be O(n x m). The network boundary is very expensive. So, in this case, you should fetch data from the DB, and then organize those data for a return value.

Reply:
In my opinion, we should avoid the scan as [the document mentioned](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-query-scan.html).
We need to call the DB three times, get groups, get container IDs and group data, and get container details.